smdebug/tensorflow/base_hook.py

```diff
@@ -422,6 +417,11 @@ def set_gradients(self, gradients=None, gradients_and_variables=None):
         :param gradients_and_variables: list of tuples [(tf.Tensor/tf.Variable, tf.Tensor/tf.Variable)...]
             list of tuples representing gradients and weights
         """
+        # TF 2.x doesn't provide gradient/optimizer variable names and values by default.
+        # Skipping set_gradients and set_optimizer_variables for Tf 2.x until there is
+        # support to pass names and values from TF side.
+        if is_tf_version_2x() and tf.executing_eagerly():
+            return
```

Comment:
most of the time 2.x will be eagerly executing and this will return true. We don't do anything in those cases?

Reply:
2.x doesn't provide grad or optimizer variable names and values by default. this function is used by zcc to pass grad and optimizer variable names, but we don't get values because TF 2.x doesn't pass values through callbacks. so set_gradients() and set_optimizer_variables() are returning without doing anything for now (in 2.x eager mode alone), until we can add support to get values.